src/commands/mod.rs

```diff
@@ -289,6 +289,58 @@ impl DaemonControl {
         GetAddressResult::new(address)
     }
 
+    /// list addresses
+    pub fn list_addresses(
+        &self,
+        start_index: Option<u32>,
+        count: Option<u32>,
+    ) -> ListAddressesResult {
+        let mut db_conn = self.db.connection();
+        let receive_index: u32 = db_conn.receive_index().into();
+        let change_index: u32 = db_conn.change_index().into();
+
+        let start_index = start_index.unwrap_or(0);
+
+        let index = if let Some(c) = count {
+            (start_index + c).saturating_sub(1)
```

Comment:
No. We shouldn't silently roll over. We should explicitly error if the start_index and/or count are so large that this goes above `u32::MAX`.

I would use checked_[add/sub] and return an error if it returns None. Something like:
```suggestion
            start_index.checked_add(c).and_then(|index| index.checked_sub(1)).flatten()
```